### PR TITLE
Ten days corridor level performance mart model

### PR DESCRIPTION
**Pull request by mmmiah**
This is the model built for the purpose of HOV Vs ML performance visualization in station level as well as corridor level. The main focus of this visualization will be Average per lane volume, vmt and vht penetration for recent 10 days as all well as am peak, pm peak, am of peak, pm off peak and night off peak temporal resolution understanding!

### CODE REVIEW

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    from station_pairs
+),
+
+closest_station_with_selection as (
+    select *
+    from closest_stations
+    where
+        distance_ranking = 1
+        and delta_postmile <= 5
+),
```
Comment (kengodleskidot) [inline transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql:83]: Is the intent to exclude stations completely from the data set that don't meet these conditions? Or do we keep all stations and then note that there is no station meeting the criteria?

**Review by kengodleskidot — COMMENTED** [review]
A few comments for your response.

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        cy.hourly_vht as hov_hourly_vht
+    from station_with_ml_metrics as ay
+    inner join hourly_station_hov_metrics as cy
+        on
+            ay.hov_station_id = cy.station_id
+            and ay.direction = cy.direction
+            and ay.sample_date = cy.sample_date
+            and ay.sample_hour = cy.sample_hour
+),
+
```
Comment (JamesSLogan) [inline transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql:140]: Unless it's less readable or more confusing, I suggest compressing the 4 above CTEs into 1 - something like this:
```sql
station_with_ml_hov_metrics as (
    select
        ax.*,
        ml.sample_date,
        ml.sample_hour,
        ml.hourly_volume as ml_hourly_volume,
        ml.hourly_vmt as ml_hourly_vmt,
        ml.hourly_vht as ml_hourly_vht,
        hov.hourly_volume as hov_hourly_volume,
        hov.hourly_vmt as hov_hourly_vmt,
        hov.hourly_vht as hov_hourly_vht
    from closest_station_with_selection ax
    inner join hourly_station_volume ml
        on ax.ml_station_id = ml.station_id
        and ax.direction = ml.direction
        and ml.station_type = 'ML'
    inner join hourly_station_volume hov
        on ax.hov_station_id = hov.station_id
        and ax.direction = hov.direction
        and ml.sample_date = hov.sample_date
        and ml.sample_hour = hov.sample_hour
        and hov.station_type = 'HV'
)
```

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        a._valid_from,
+        a._valid_to
+    from mainline as a
+    inner join hov as b
+        on
+            a.freeway = b.freeway
+            and a.direction = b.direction
+            and a._valid_from = b._valid_from
+            and a.ml_station_id != b.hov_station_id
+),
```
Comment (JamesSLogan) [inline transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql:68]: Would it be possible to leverage the `int_vds__nearby_stations` model for this? Either way, the `mainline` and `hov` CTEs could be collapsed into this one, something like:

```sql
station_pairs as (
    select
        <columns>
    from station_meta ml
    inner join station_meta hov
    on
        ml.freeway = hov.freeway
        and ml.direction = hov.direction
        and ml._valid_from = hov._valid_from
        and ml.station_id != hov.station_id
        and ml.station_type = 'ML'
        and hov.station_type = 'HV'
)
```

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql (hunk 1 above)
Comment (JamesSLogan) [inline transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql:83]: Suggest eliminating the `closest_stations` CTE through use of QUALIFY:
```sql
closest_station_with_selection as (
    select
        *,
        row_number() over (partition by ml_station_id, _valid_from order by delta_postmile asc) as distance_ranking
    from station_pairs
    where
        delta_postmile <= 5
    qualify
        distance_ranking = 1
),
```

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql (hunk 3 above)
Comment (mmmiah) [reply transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql:38]: Thanks! James! Incorporated your all great suggestions!

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql (hunk 1 above)
Comment (mmmiah) [reply transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql:49]: included a qualify function and removed previous CTE

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql (hunk 2 above)
Comment (mmmiah) [reply transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql:90]: updated the code with your proposed one

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        station_id,
+        direction,
+        sample_date,
+        sample_hour,
+        hourly_volume,
+        hourly_vmt,
+        hourly_vht,
+        station_type
+    from {{ ref('int_performance__station_metrics_agg_hourly') }}
+    where sample_date = dateadd('day', -25, current_date())
```
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql]: my model was reading data from transform dev which did not get data for last 10 tens, I have dropped the table from transform dev, now I set it 12 days back from current days!

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+final_data_with_catagory as (
+    select
+        station_metric_agg.*,
+        case
+            when extract(hour from station_metric_agg.sample_hour) between 6 and 8 then 'Morning Peak'
+            when extract(hour from station_metric_agg.sample_hour) between 9 and 11 then 'Morning Off Peak'
+            when extract(hour from station_metric_agg.sample_hour) between 12 and 14 then 'Afternoon Off Peak'
+            when extract(hour from station_metric_agg.sample_hour) between 15 and 18 then 'Afternoon Peak'
+            else 'Night Off Peak'
```
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql]: This is great suggestions! I have set the variables as you suggested

File: transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql (hunk 1 above)
Comment (mmmiah) [reply transform/models/marts/performance/performance__station_metric_agg_recent_ten_days.sql:49]: We are not keeping all those main stations for now, as that will be zero value for vmt, vht and volume penetration as there is no HOV detector against each main lines.

**Review by JamesSLogan — APPROVED** [review]
No further comments from me 👍

**Comment by kengodleskidot** [comment]
No additional comments from me @mmmiah 👍

**Pull request merged**